src/main/java/io/pinecone/clients/AsyncIndex.java

```diff
@@ -216,6 +225,7 @@ public ListenableFuture<DescribeIndexStatsResponse> describeIndexStats(Struct fi
     public void close() {
         try {
             logger.debug("closing channel");
+            pinecone.closeConnection(indexName);
             connection.getChannel().shutdownNow().awaitTermination(5, TimeUnit.SECONDS);
         } catch (InterruptedException e) {
```

Comment:
Rather than get the channel from the connection and shutting it down, can't we just call `connection.close()`?

Reply:
Yes, totally!!